### PR TITLE
W-11660023-database-xa-transactions-kt

### PR DESCRIPTION
**Pull request by kevintroller**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: db/1.13/modules/ROOT/pages/database-connector-connection.adoc [hunk 1]
```diff
@@ -61,6 +61,7 @@ In the XML editor, both the spring and database configurations look like this:
 </db:config>
 ----
 
+[[data-source-connection-xa]]
 == Configure a Data Source Reference Connection When in XA Transaction Mode
 
 If your Mule app requires operations bound to XA transactions, and connection pooling is enabled for the database configuration in use, the default connection pool implementation for Data Source Reference Connection is Enhydra XAPool.
```
Comment (nicomz) [inline db/1.13/modules/ROOT/pages/database-connector-connection.adoc:67]: As far as I remember, Enhydra is the underlying datasource. The pooling datasource is the one provided by Bitronix.

**Review by nicomz — DISMISSED** [review]

**Review by acurci-at-mulesoft — DISMISSED** [review]
What do you think about mentioning that when an app uses a XA transaction the datasource will be an instance of Enhydra datasource?

File: db/1.13/modules/ROOT/pages/database-connector-connection.adoc (hunk 1 above)
Comment (acurci-at-mulesoft) [reply db/1.13/modules/ROOT/pages/database-connector-connection.adoc:67]: Enhydra has a XAPool also, but I think we are not using it. We are only using the XA datasource implementations. I'd only mention about datasource implementation.

Comment (kevintroller) [reply db/1.13/modules/ROOT/pages/database-connector-connection.adoc:67]: Hi guys @nicomz @acurci-at-mulesoft , I'm a bit lost, so if the user selects Data Source Reference connection, what is the default connection pool then? See the following section and example after that sentence:

_Enhydra XAPool does not support prepared statements. To execute these operations use the Bitronix connection pool. Refer to xref:mule-runtime::using-bitronix-to-manage-transactions.adoc#configuring-the-bitronix-transaction-manager[Configuring the Bitronix Transaction Manager] documentation for details._

_The following example shows how to configure a *Data Source Reference Connection* for a Derby database when in XA transaction mode:
. In Studio, in the `src/main/resources` folder, create a `datasource-config.xml` spring bean file.
. Add the `XADataSource` object to the end of the class `org.enhydra.jdbc.standard.Standard`, for example, `class="org.enhydra.jdbc.standard.StandardXaDataSource`:_

File: db/1.13/modules/ROOT/pages/database-connector-xa-transactions.adoc [hunk 2]
```diff
@@ -0,0 +1,44 @@
+= Using XA Transactions in Database Connector
+
+Anypoint Connector for Database (Database Connector) provides the option to use Extended Architecture Transactions (XA Transactions) when you enable the *Use XA Transactions* field in the Database Connector global configuration.
+
+== Before You Begin
+
+Before getting started, ensure that you have understanding of xref:mule-runtime::xa-transactions.adoc[XA Transactions]
```
Comment (dhirsch1) [inline db/1.13/modules/ROOT/pages/database-connector-xa-transactions.adoc]: Before getting started, ensure that you understand xref:mule-runtime::xa-transactions.adoc[XA Transactions].

or

Before getting started, ensure that you have an understanding of xref:mule-runtime::xa-transactions.adoc[XA Transactions].

File: db/1.13/modules/ROOT/pages/database-connector-xa-transactions.adoc [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+== Use XA Transactions
+
+You can enable the *Use XA Transactions* field in the database global connector configuration for the Derby Connection, Generic Connection, Microsoft SQL Server Connection, MySQL Connection and Oracle Connection.
```
Comment (dhirsch1) [inline db/1.13/modules/ROOT/pages/database-connector-xa-transactions.adoc]:
```suggestion
You can enable the *Use XA Transactions* field in the database global connector configuration for the Derby Connection, Generic Connection, Microsoft SQL Server Connection, MySQL Connection, and Oracle Connection.
```

File: db/1.13/modules/ROOT/pages/database-connector-xa-transactions.adoc [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+For a DataSource connection, refer to the xref:database-connector-connection.adoc#data-source-connection-xa[Configure a Data Source Reference Connection When in XA Transaction Mode] documentation.
```
Comment (dhirsch1) [inline db/1.13/modules/ROOT/pages/database-connector-xa-transactions.adoc]: DataSource > Data Source Reference Connection (the actual connection name in studio)

File: db/1.13/modules/ROOT/pages/database-connector-xa-transactions.adoc [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+== Use XA Transactions
+
+You can enable the *Use XA Transactions* field in the database global connector configuration for the Derby Connection, Generic Connection, Microsoft SQL Server Connection, MySQL Connection and Oracle Connection.
+For a DataSource connection, refer to the xref:database-connector-connection.adoc#data-source-connection-xa[Configure a Data Source Reference Connection When in XA Transaction Mode] documentation. 
+
+[NOTE]
+When you enable *Use XA Transactions* the connection pooling is created as well. 
+
+The following example shows how to configure *Use XA Transactions* for *MySQL Connection* in Anypoint Studio:
```
Comment (dhirsch1) [inline db/1.13/modules/ROOT/pages/database-connector-xa-transactions.adoc]:
```suggestion
The following example shows how to configure *Use XA Transactions* for the *MySQL Connection* in Anypoint Studio:
```

Comment (dhirsch1) [reply db/1.13/modules/ROOT/pages/database-connector-xa-transactions.adoc]: You didn't bold the connection names earlier - either way, but make them consistent

File: db/1.13/modules/ROOT/pages/database-connector-xa-transactions.adoc [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+. In Studio, navigate to the *Global Elements* tab.
+. Click *Create*.
+. In the search bar, type `database`.
+. Select *Database Config*.
+. Click *OK*.
+. In the *Global Element Properties* window, for *Connection* select *MySQL Connection*.
```
Comment (dhirsch1) [inline db/1.13/modules/ROOT/pages/database-connector-xa-transactions.adoc]: Suggest: In the *Global Element Properties* window, select *MySQL Connection* in the
*Connection* field.

File: db/1.13/modules/ROOT/pages/database-connector-xa-transactions.adoc [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+. Select the *Transactions* tab.
+. Select *Use XA Transactions* that indicates that the created data source must support XA transactions. +
```
Comment (dhirsch1) [inline db/1.13/modules/ROOT/pages/database-connector-xa-transactions.adoc]:
```suggestion
. Select *Use XA Transactions*, which indicates that the created data source must support XA transactions. +
```

File: db/1.13/modules/ROOT/pages/database-connector-examples.adoc [hunk 8]
```diff
@@ -31,6 +31,9 @@ Configure custom data types to use when connected to a particular connection pro
 Execute database operations in the context of a transaction.
 * xref:database-insert-large-clob.adoc[Insert large file content into a table] +
 Configure a Mule app flow to read and insert a large file into a table.
+* xref:database-connector-xa-transactions.adoc[Using XA Transactions in Database Connector] +
+Configure the *Use XA Transactions* field in the Database Connector global configuration so the datasource supports Extended Architecture transactions.
```
Comment (dhirsch1) [inline db/1.13/modules/ROOT/pages/database-connector-examples.adoc]: Later on, you have "Extended Architecture Transactions (XA Transactions)." If this is the first mention in the topic, you might want to put that here as well.

File: db/1.13/modules/ROOT/pages/database-connector-xa-transactions.adoc (hunk 4 above)
Comment (dhirsch1) [reply db/1.13/modules/ROOT/pages/database-connector-xa-transactions.adoc]: documentation > topic ???

**Review by dhirsch1 — DISMISSED** [review]

**Review by dhirsch1 — APPROVED** [review]

**Pull request merged**